runtime/gc_vlhgc/CopyForwardScheme.cpp

```diff
@@ -1452,6 +1463,92 @@ MM_CopyForwardScheme::mainSetupForCopyForward(MM_EnvironmentVLHGC *env)
 	_shouldScanFinalizableObjects = _extensions->finalizeListManager->isFinalizableObjectProcessingRequired();
 }
 
+void
+MM_CopyForwardScheme::sortAllHotFieldData()
```

Comment:
We should share this sorting code with Scavenger(Delegate). 
We could also move it to core VM, and pass a few GC specific parameters from GC Extensions through method arguments.

@dmitripivkine do you have a preference if this should stay in GC or VM?

Reply:
I agree this code should be shared between GC policies. We can strip out core functionality and leave policy dependable code (like stats update) in the place.
I think we can keep this code in GC for now. We can put it in `gc_base`. We can make this code more independent from other GC code by replacing direct references to GC Extensions (like `_extensions->depthCopyThreePaths`) to parameters. This will make move to VM easier if we need it in the future.

Reply:
ok, let's create gc_base/HotFieldUtil.hpp/cpp and move those 2 sorting methods from ScavDelegate there

Reply:
I do also certainly agree this code should be shared as much as possible between policies. However, although a large portion of the sorting methods are the same, there are differences between the sorting methods in ScavengerDelegate and CopyForwardScheme unique to the  gc policy which is what prevented me to share this sorting code in a cleaner manner (balanced does not use _extensions->allowPermanantHotFields and _extensions->adaptiveGcCountBetweenHotFieldSort). The only method that could be easily shared was resetAllHotFieldData which I moved to the VM. The core elements of the sorting methods are the same and could be shared. With that said, how do you recommend I proceed? Move the resetAllHotFieldData and core elements of the hot field sorting methods to gc_base, perhaps modronapi.cpp? I believe this would allow for max code sharing and I don't believe it would affect performance. Whatever you both decide I can gladly do this.  @dmitripivkine @amicic